### PR TITLE
Improved log performance and reliability

### PR DESCRIPTION
**Pull request by jeffdm**
For performance reasons, only allow attaching to one `active` container at a time. This container is set to the one being viewed by the user.

### CODE REVIEW

**Comment by mchiang0610** [comment]
LGTM. Works very well on Windows!

File: src/stores/ContainerStore.js [hunk 1]
```diff
@@ -158,6 +158,35 @@ class ContainerStore {
...
+  logs ({name, logs}) {
+    let container = this.containers[name];
+
+    if (!container) {
+      return;
+    }
+
+    container.Logs = logs.split('\n');
+
+    console.log(container.Logs);
```
Comment (FrenchBen) [inline src/stores/ContainerStore.js]: Can we remove this debug?

File: src/utils/DockerUtil.js [hunk 2]
```diff
@@ -341,16 +422,24 @@ export default {
       stream.on('data', json => {
         let data = JSON.parse(json);
 
-        if (data.status === 'pull' || data.status === 'untag' || data.status === 'delete' || data.status === 'attach') {
+        console.log('Event from daemon: ' + data.status + ' ' + data.id);
```
Comment (FrenchBen) [inline src/utils/DockerUtil.js]: Do we want to keep this debug/log ?

File: src/utils/DockerUtil.js [hunk 3]
```diff
@@ -330,6 +332,85 @@ export default {
...
+      logStream.setEncoding('utf8');
+      logStream.on('data', chunk => logs += chunk);
+      logStream.on('end', () => {
+        containerServerActions.logs({name: this.activeContainerName, logs});
+        this.attach();
+      });
+    });
+  },
+
+  attach () {
```
Comment (FrenchBen) [inline src/utils/DockerUtil.js]: Seems that attach() and logs() do very similar things, could they be merged into 1 method?

**Comment by FrenchBen** [comment]
A few comments but otherwise looks awesome!

File: src/utils/DockerUtil.js (hunk 3 above)
Comment (jeffdm) [inline src/utils/DockerUtil.js]: They are similar in that they both call the /logs api endpoint, but they behave differently - one streams changes and the other hooks into the logs. I tried merging them in various ways, but kept leaning towards two functions.

**Comment by FrenchBen** [comment]
LGTM

**Comment by FrenchBen** [comment]
:+1: All looks good

**Pull request merged**